src/cordova/build.action.mjs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import cordovaLib from "cordova-lib";
+const {cordova} = cordovaLib;
+
+import {getBuildParameters} from "../../scripts/get_build_parameters.mjs";
+
+export const requirements = ["cordova/setup"];
+
+/**
+ * @description Builds the parameterized cordova binary (ios, osx, android).
+ * WARNING: still a WIP.
+ *
+ * @param {string[]} parameters
+ */
+export async function main(...parameters) {
+  const {platform, buildMode} = getBuildParameters(parameters);
+
+  let downstreamArgv = [];
+
+  if (platform === "android") {
+    downstreamArgv.push("--gradleArg=-PcdvBuildMultipleApks=true");
+  }
+
+  if (
+    platform === "android" &&
+    buildMode === "release" &&
+    !(process.env.ANDROID_KEY_STORE_PASSWORD && process.env.ANDROID_KEY_STORE_CONTENTS)
+  ) {
+    throw new ReferenceError(
+      "Both 'ANDROID_KEY_STORE_PASSWORD' and 'ANDROID_KEY_STORE_CONTENTS' must be defined in the environment to build an Android Release!"
+    );
+  }
+
+  await cordova.compile({
+    platforms: [platform],
+    options: {
+      device: platform === "ios",
+      release: buildMode === "release",
+      argv:
```

Comment:
This was derived by crawling the cordova source until I got to this:

https://github.com/apache/cordova-cli/blob/7b0e24ebc3439c0daabcf2995b964023c57dcc19/src/cli.js#L381

Reply:
I know I know you're not gonna like it, `cordova-lib` is not well documented 😅